### ISSUE TITLE
Feature request: Add option to create ZIP file of backup directory

### ISSUE THREAD

**Issue by renatobellotti** [machiav3lli/oandbackupx#447]
# Idea

Add a checkbox "ZIP after backup" and a user defined ZIP path.

# Reason

Having a regular backup is only part of a reasonable backup strategy. The other part is getting the backup off the device. I'm currently syncing the backup directory to my Nextcloud server, which takes very long (much longer than 1h). I suspect that Nextcloud struggles when there are many small files. Having one big ZIP file to sync makes it easier to create off-device backups.

**Comment by hg42** [comment]
From my experience nextcloud is slow in any way.
You may better use syncthing.

A big zip doesn't look like a good solution.
We shouldn't add workarounds for problems of other software.

just my opinion...

**Comment by renatobellotti** [comment]
I respectfully disagree here. IMHO the problem is much more general: Any network connection can be sped up by converting the latency problem into a bandwidth problem. However, sending many small files always introduces additional latency.

Having said this, is there a recommended way to get the backup off the device? I think plugging it in regularly is too manual to be a true alternative to popular backups that run in the background. But again, this might just be me.

**Comment by JohnCGH** [comment]
What about doing something like what Titanium Backup did.....Allow a user to tap on a backup and select "Share".

It would then zip up all files in the backup into one ZIP file, then prompt the user:

1) To send a regular intent to OS to display the typical "Share" menu, so for example the user can "Email" the zip file.
2) Allow the user to save the ZIP file to a local directory.

(I use https://freefilesync.org/ to sync the whole backup directory inside the phone to a directory on my PC using a usb cable)

**Comment by machiav3lli** [comment]
The both mentioned features are on my todo list but go with lowest priority. As always contributing low priority features is a welcome thing.

**Comment by Mitezuss** [comment]
@machiav3lli 

Hi, can add to some feacture to "import" a zip ? previusly backup exported as zip file

Or, can just make a tar (only store) of the folder of backup ? instead of the folders (into the backup folder)
